lms/djangoapps/course_home_api/tasks.py

```diff
@@ -37,12 +37,19 @@ def collect_progress_for_user_in_course(course_id: str, user_id: str) -> None:
         log.warning(f"Could not retrieve a user with id {user_id}, aborting task.")
         return
 
+    try:
+        enrollment = get_course_enrollment(user, course_key)
+        enrollment_mode = enrollment.mode
+    except AttributeError:
+        log.warning(f"Could not retrieve enrollment info for user {user.id} in course {course_id}")
+        return
```

Comment:
I opted to send nothing here. My feelings were... if we can't retrieve the enrollment track, then there is too much ambiguity to emit the event. Happy to discuss further.